### ISSUE TITLE
`os.path.dirname` throws `UnicodeDecodeError` on Windows

### ISSUE THREAD

**Issue by GCS-ZHN** [xapple/autopaths#2]
When try to create a file in working directory without "./", Will result a UnicodeDecodeError.

OS: Windows 11
Python: 3.9.7

```python
FilePath("test.txt").directory;
FilePath("test.txt").create();
```

When you trackback the error, you can found it resulted from following code:
```python
if not directory: directory = os.path.dirname(self.absolute_path)
```

Based on  your definition, `self.absolute_path` will return a `FilePath` object, 
but `os.path.dirname` requires for `str` type argument.

I don't think it is necessary to wrap property `self.absolute_path` and `self.physical_path` as  `FilePath` object, 
as `self.path` already is a `str` object.

And this problem will influence downstream development such `fasta` which is also developed by yourself.

**Comment by xapple** [comment]
Hi Gcs,
Thanks for opening the issue and reporting this bug. I'm unfortunately unable to reproduce it. The following code works without any issues on both macOS and Ubuntu. I don't have access to a Windows machine however.

    (base) xapple@ubuntu test $ ipython

    In [1]: import autopaths

    In [2]: from autopaths.file_path import FilePath

    In [3]: p = FilePath('test.txt')

    In [4]: p.directory
    Out[4]: <DirectoryPath object "/home/xapple/test/">

    In [5]: p.create()
    Out[5]: <FilePath object "test.txt">

    In [6]:

**Comment by GCS-ZHN** [comment]
邮件已收到，祝您身体健康，工作顺利！

**Comment by GCS-ZHN** [comment]
Because `os.path.dirname`  have different behavours on different OS, on windows, a non-str or non-bytes object arguments not automatically be converted as `str` path? So I just recommand you to return a `str` object on `self.absolute_path` property to avoid this difference.

when try it on my ubuntu server, it's ok, but on windows, a error raised.
```python
> import os
> from autopaths.file_path import FilePath
> os.path.dirname(FilePath("test.py"))
UnicodeDecodeError                        Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_32928/854867784.py in <module>
      1 import os
----> 2 os.path.dirname(FilePath("test.py"))

c:\ProgramData\Anaconda3\lib\ntpath.py in dirname(p)
    221 def dirname(p):
    222     """Returns the directory component of a pathname"""
--> 223     return split(p)[0]
    224 
    225 # Is a path a symbolic link?

c:\ProgramData\Anaconda3\lib\ntpath.py in split(p)
    185     p = os.fspath(p)
    186     seps = _get_bothseps(p)
--> 187     d, p = splitdrive(p)
    188     # set i to index beyond p's last slash
    189     i = len(p)

c:\ProgramData\Anaconda3\lib\ntpath.py in splitdrive(p)
    142     """
    143     p = os.fspath(p)
--> 144     if len(p) >= 2:
    145         if isinstance(p, bytes):
...
---> 45         msg = msg % (full_cmd, out.decode(), err.decode())
     46         # Call parent #
     47         super(ErrorReturnCode, self).__init__(msg)

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbe in position 0: invalid start byte

```

**Comment by xapple** [comment]
Yes, it seems that when you the user is on windows, `path.py` is actually `ntpath.py` which can be seen here:

https://github.com/python/cpython/blob/main/Lib/ntpath.py

But, I'm still not sure what is going on exactly, do you think you could post the full traceback?

**Comment by xapple** [comment]
One simple change could be to wrap the input argument of all the calls to `os.path` with `str()`. This would be a better solution as compared to changing the return type of the methods in the library. Would you like to submit a pull request and maybe a unittest or two?

**Comment by xapple** [comment]
I have too few resources for this project, so supporting Windows is unfortunately not a priority. I will be closing this issue for now.
